src/pages/Signup/Signup.js

```diff
@@ -1,52 +1,139 @@
-import React from 'react'
-import './signup.css'
+import React, { useState } from "react";
+import "./signup.css";
+import axios from "axios";
+import { useNavigate } from "react-router-dom";
+import { useDispatch } from "react-redux";
+import { setCurrentUser, signupError } from "../../slice/signupslice";
 
+const url = "http://localhost:4000/v1/signup";
```

Comment:
You shouldn't hardcode your API URL. It should be dynamic based on the environment you are on. When you deploy this code online, your API URL will no longer be localhost:3006, how will you solve the issue in that case?